src/vmemcache_repl.c

```diff
@@ -267,30 +309,36 @@ static void *
 repl_p_lru_evict(struct repl_p_head *head, struct repl_p_entry **ptr_entry)
 {
 	struct repl_p_entry *entry;
-	void *data;
+	void *data = NULL;
 
 	util_mutex_lock(&head->lock);
 
-	if (ptr_entry != NULL)
-		entry = *ptr_entry;
-	else
-		entry = TAILQ_FIRST(&head->first);
+	if (TAILQ_EMPTY(&head->first)) {
+		errno = ESRCH;
+		goto exit_unlock;
+	}
+
+	if (ptr_entry == NULL)
+		ptr_entry = TAILQ_FIRST(&head->first)->ptr_entry;
+
+	int max_retry = 2;
+
+	while ((entry = util_val_compare_and_swap64(ptr_entry,
+			*ptr_entry, NULL)) == NULL && (--max_retry > 0)) {
```

Comment:
same question as in repl_p_lru_use

Reply:
I do care. 'entry' is the previous value.